src/test/java/org/openelisglobal/storage/controller/StorageLocationRestControllerTest.java

```diff
@@ -0,0 +1,197 @@
+package org.openelisglobal.storage.controller;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import javax.sql.DataSource;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.openelisglobal.BaseWebContextSensitiveTest;
+import org.openelisglobal.login.dao.UserModuleService;
+import org.openelisglobal.storage.service.StorageLocationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.util.ReflectionTestUtils;
+
+/**
+ * Controller integration tests for Storage Location CRUD endpoints.
+ */
+@RunWith(SpringRunner.class)
+public class StorageLocationRestControllerTest extends BaseWebContextSensitiveTest {
+
+    @Autowired
+    private StorageLocationService storageLocationService;
+
+    @Autowired
+    private DataSource dataSource;
+
+    @Autowired
+    private StorageLocationRestController storageLocationRestController;
+
+    private UserModuleService userModuleServiceMock;
+
+    private ObjectMapper objectMapper;
+    private JdbcTemplate jdbcTemplate;
+
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        objectMapper = new ObjectMapper();
+        jdbcTemplate = new JdbcTemplate(dataSource);
+        cleanTestData();
+        userModuleServiceMock = Mockito.mock(UserModuleService.class);
+        ReflectionTestUtils.setField(storageLocationRestController, "userModuleService", userModuleServiceMock);
+        when(userModuleServiceMock.isUserAdmin(any())).thenReturn(true);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        cleanTestData();
+    }
+
+    private void cleanTestData() {
+        try {
+            jdbcTemplate.execute("DELETE FROM storage_device WHERE id >= 20000");
+            jdbcTemplate.execute("DELETE FROM storage_shelf WHERE id >= 20000");
+            jdbcTemplate.execute("DELETE FROM storage_rack WHERE id >= 20000");
+            jdbcTemplate.execute("DELETE FROM storage_room WHERE id >= 20000");
+        } catch (Exception e) {
+            // Ignore cleanup errors
+        }
+    }
+
+    @Test
+    public void testDeleteRoom_WithValidId_Returns204() throws Exception {
+        // Arrange - Create room with no children
+        jdbcTemplate.update(
```

Comment:
we prefer using dbUnit xml datasets for cleaner and more easiliy mantainable Tests , but we can refactor later